src/tcp_interface.cpp

```diff
@@ -74,7 +74,7 @@ ReturnStatuses TCPInterface::read(std::vector<uint8_t> *msg)
 
   error_.assign(boost::system::errc::success, boost::system::system_category());
 
-  msg->reserve(10000);
+  msg->resize(socket_.available(), 0);
```

Comment:
Just so I'm understanding correctly, this was never an issue before because the amount of bytes to read was always equal to or less than 10000? The ibeo use case has revealed this bug because the amount of data is so high?

Reply:
No, this wasn't working at all before this change. The reason was that I misunderstood what `boost::asio::read` was expecting in terms of a buffer. Since it accepts a vector in addition to a fixed-size buffer, I assumed that it would create the necessary objects in the vector to store the received data which is why I used `reserve()` which only allocates the memory storage and does not create the actual objects in the vector. Using `resize()` instead actually creates the provided number of objects in the vector.